lib/mix/lib/mix/tasks/test.ex

```diff
@@ -556,7 +557,7 @@ defmodule Mix.Tasks.Test do
 
   defp filter_to_allowed_files(matched_test_files, nil), do: matched_test_files
 
-  defp filter_to_allowed_files(matched_test_files, %MapSet{} = allowed_files) do
```

Comment:
`MapSet.t()` is a opaque type which means that we cannot pattern match on this value

Reply:
I would honestly keep doing this check. This is a limitation of Dialyzer-Elixir integration and I wouldn't want to make the code less precise only to satisfy Dialyzer.